data_access/eth_chain.py

```diff
@@ -760,23 +760,22 @@ def tokens(self, web3=None):
         """Returns a list of ERC20 tokens supported by the Well."""
         return call_contract_function_with_retry(self.contract.functions.tokens())
 
-    def get_beans_sent(self, txn_hash):
-        """Return the amount (as a float) of BEAN sent in a transaction"""
-        log = get_erc20_transfer_log_in_txn(BEAN_ADDR, txn_hash)
-        if log:
-            return int(log.data, 16)
-        return 0
+    def get_beans_sent(self, txn_hash, recipient, log_end_index):
+        """Return the amount (as a float) of BEAN sent in a transaction to the given recipient, prior to the provided log index"""
+        logs = get_erc20_transfer_logs_in_txn(BEAN_ADDR, txn_hash, recipient, log_end_index)
+        total_sum = 0
+        for entry in logs:
+            total_sum += int(entry.data, 16)
+        return total_sum
 
 
-def get_eth_sent(txn_hash, web3):
-    """Return the amount (as a float) of ETH or WETH sent in a transaction"""
-    txn_value = web3.eth.get_transaction(txn_hash).value
-    if txn_value != 0:
-        return txn_value
-    log = get_erc20_transfer_log_in_txn(WRAPPED_ETH, txn_hash, web3=web3)
-    if log:
-        return int(log.data, 16)
-    return 0
+def get_eth_sent(txn_hash, recipient, web3, log_end_index):
+    """Return the amount (as a float) of ETH or WETH sent in a transaction to the given recipient, prior to the provided log index"""
```

Comment:
this is no longer _OR_ behavior. it is now a sum. i cannot recall why it was OR previously